app/workers/deletes.php

```diff
@@ -389,9 +389,26 @@ protected function getConsoleDB(): Database
     {
         global $register;
 
-        $cache = new Cache(new RedisCache($register->get('cache')));
-        $dbForConsole = new Database(new MariaDB($register->get('db')), $cache);
-        $dbForConsole->setNamespace('project_console_internal'); // Main DB
+        // wait for database to be ready
+        $attempts = 0;
```

Comment:
A few questions:

Why do we have this logic only on this worker?
Should we apply to other workers? 
Should we apply this logic on the parent class?
Should we also apply it on `getExternalDB` and `getInternalDB`?

Reply:
I included this logic in the deletes worker because it's the only worker that requires a database connection near instantly on startup.
These are the logs of the deletes worker without this logic - you can see that it receives jobs to clean up executions, audit logs, and abuse logs by timestamp almost instantly, before the database is ready:

https://gist.github.com/kodumbeats/a8b80f22f09ecb733802702c28c10628

These three delete jobs need the consoleDB to search for projects here:
https://github.com/appwrite/appwrite/blob/8dab66d455a7d986847bec8edb39029ae73b4d1a/app/workers/deletes.php#L275

So, we only need the try-catch around the consoleDB for the deletes worker to confirm zero projects.

Reply:
I think that now that we have a share [Worker class](https://github.com/appwrite/appwrite/blob/master/src/Appwrite/Resque/Worker.php) that all the worker extend we can move all the DB getters to there. It's also an Appwrite class so I think coupling it with the new DB classes is not as major issue as usually treat it to be.

If we have the retry logic on one DB adapter I don't see any reason why we shouldn't keep consistency and avoid future issues by implementing this logic on all the db getters (getConsoleDB, getInternalDB, getExternalDB)